dist/turtler.js

```diff
@@ -15,43 +15,46 @@ module.exports = function turtler(data, options={}) {
   let table = '';
   let columns = 0;
   let columnWidths = [];
-  let { hasHeader=true, columnSeperator=' | ', headerSeperator='=' } = options;
+  let { hasHeader=true, columnSeparator=' | ', headerSeparator='=' } = options;
 
   // Find the maximum width of each column
-  // If a column contains an odd number of values throw
-  for(var i = 0; i < data.length; i++) {
-    let line = data[i];
+  // If rows contain uneven number of columns, throw
+  data.forEach((row) => {
+    // The row should be an array
+    if(!Array.isArray(row)) throw new Error('data should be an array of arrays');
+    // Set the initial length of the row
+    if(!columns) columns = row.length;
+    // If the current row is not the same length as the initial one throw error
+    if(columns !== row.length) throw new Error('columns are not formed properly');
 
-    if(!Array.isArray(line)) throw new Error('data should be an array of arrays');
-    if(!columns) columns = line.length;
-    if(columns !== line.length) throw new Error('columns are not formed properly');
-
-    // find the maximum length of the column
-    line.forEach((v, l) => {
+    // find the maximum length of each column
+    row.forEach((v, l) => {
+      // column values must be strings
       if(typeof v !== 'string') throw new Error('column values should be strings');
 
+      // Find the maximum string length in each column
       if(!columnWidths[l] || columnWidths[l] < v.length) {
         columnWidths[l] = v.length;
       }
     });
-  }
+  });
 
-  data.forEach((line, l) => {
-    let row = line.map((column, i) => {
-      let value = '';
-      for(var c = 0; c < columnWidths[i]; c++) {
-        value += column[c] || ' ';
-      }
-      return value;
-    }).join(columnSeperator);
+  data.forEach((row, l) => {
+    row = row.map((value, i) => {
+      // Create pad of empty spaces to match the width of this value to max width of this column
+      let padding = ' '.repeat(columnWidths[i] - value.length);
```

Comment:
Does repeat use a for loop internally? Creating a padded string and appending it will cause more operations then altering a single string object right?

Reply:
[Spec](http://www.ecma-international.org/ecma-262/6.0/#sec-string.prototype.repeat) says 
```
Let T be a String value that is made from n copies of S appended together. If n is 0, T is the empty String.
```

[Source code](https://github.com/nodejs/node/blob/7940ecfa0093d1cc5b793f59a5c48b5050211d6b/deps/v8/src/js/string.js#L688-L693) shows a loop.

If we're nitpicky about performance, *_technically_* the proposed change is slightly more efficient, since it doesn't go over the source string, and doesn't do an OR operation, so it saves those cycles.

However, I made the change for readability

Reply:
Perfect, just wanted to make sure. I was going to use repeat, but the last I looked it was a little bit more efficient, but the readability is great! Thanks!